airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py

```diff
@@ -473,7 +476,69 @@ def request_params(
         return params
 
 
-class SourceZendeskIncrementalExportStream(SourceZendeskSupportCursorPaginationStream):
+class SourceZendeskSupportCursorPaginationStream(SourceZendeskSupportFullRefreshStream):
+    """
+    Only used by streams modified by rudderstack
+    Implements cursor pagination as most streams support cursor pagination.
+    Where it is not supported such streams will override some of this functionality to
+    use offset pagination.
+    """
+
+    cursor_field = "updated_at"
+    next_page_field = "next_page"
+    prev_start_time = None
+
+    @property
+    def state_checkpoint_interval(self) -> Optional[int]:
+        """
+        Will allow the connector send state messages more frequent and not only at the end of the sync.
+        """
+        return 1000
+
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
+        # try to save maximum value of a cursor field
+        old_value = str((current_stream_state or {}).get(self.cursor_field, ""))
+        new_value = str((latest_record or {}).get(self.cursor_field, ""))
+        return {self.cursor_field: max(new_value, old_value)}
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        if self._ignore_pagination:
+            return None
+        """
+        https://developer.zendesk.com/documentation/api-basics/pagination/paginating-through-lists-using-cursor-pagination/#when-to-stop-paginating
+        """
+        meta = response.json().get("meta", {})
+        return {"page[after]": meta.get("after_cursor")} if meta.get("has_more") else None
+
+    def check_stream_state(self, stream_state: Mapping[str, Any] = None):
+        """
+        Returns the state value, if exists. Otherwise, returns user defined `Start Date`.
+        """
+        state = stream_state.get(self.cursor_field) or self._start_date if stream_state else self._start_date
+        return calendar.timegm(pendulum.parse(state).utctimetuple())
```

Comment:
Could not find proper details in API ref doc of Zendesk.

So cursor field values are not dates but some long string value like `eyJvIjoiLWNyZWF0ZWRfYXQsLWlkIiwidiI6IlpFMkRubDBBQUFBQWFVTG1QcDFVQUFBQSJ9`

How is state date say at start vs this value? And how are we able to use `calendar.timegm(pendulum.parse(state).utctimetuple())`?

Reply:
I think I am a bit confused. Let me write what I understood from code:
1. cursor_field is updated_at (wherever applicable)
2. cursor pagination will happen on page after value like `eyJvIjoiLWNyZWF0ZWRfYXQsLWlkIiwidiI6IlpFMkRubDBBQUFBQWFVTG1QcDFVQUFBQSJ9`
3. cursor state in our DB as checkpoint would be updated_at value

In next sync, will it start from last saved value of updated_at? This will also clarify my previous [comment](https://github.com/rudderlabs/airbyte/pull/138#discussion_r1268156420)